### PR TITLE
Add Unstructured file converter

### PR DESCRIPTION
**Pull request by anakin87**
- I refactored a bit the old project of mine
- Added some basic tests

### CODE REVIEW

File: components/file_converters/unstructured_fileconverter/docker-compose.yml [hunk 1]
```diff
@@ -0,0 +1,11 @@
...
+  unstructured-api:
+    image: "quay.io/unstructured-io/unstructured-api:latest"
+    ports:
+      - 8000:8000
+    restart: on-failure
+    healthcheck:
+        test: curl --fail http://localhost:8000/healthcheck || exit 1
+        interval: 10s
+        timeout: 1s
+        retries: 10
```
Comment (masci) [inline components/file_converters/unstructured_fileconverter/docker-compose.yml:11]: do we need the compose file? I see in the readme we give instructions for plain `docker run ...`

File: components/file_converters/unstructured_fileconverter/src/unstructured_fileconverter_haystack/fileconverter.py [hunk 2]
```diff
@@ -0,0 +1,180 @@
...
+            self,
+            api_url=self.api_url,
+            document_creation_mode=self.document_creation_mode,
+            separator=self.separator,
+            unstructured_kwargs=self.unstructured_kwargs,
+            progress_bar=self.progress_bar,
+        )
+
+    @classmethod
+    def from_dict(cls, data: Dict[str, Any]) -> "UnstructuredFileConverter":
```
Comment (masci) [inline components/file_converters/unstructured_fileconverter/src/unstructured_fileconverter_haystack/fileconverter.py:87]: you can omit this one if all it does is calling `default_from_dict`

**Review by masci — CHANGES_REQUESTED** [review]

**Comment by masci** [comment]
Almost forgot, can you also update the repo README?

File: components/file_converters/unstructured_fileconverter/docker-compose.yml (hunk 1 above)
Comment (anakin87) [reply components/file_converters/unstructured_fileconverter/docker-compose.yml:11]: It is used to run the tests. I would prefer to update the readme.

Comment (masci) [reply components/file_converters/unstructured_fileconverter/docker-compose.yml:11]: Why not using workflow `services`? See https://github.com/deepset-ai/haystack/blob/main/.github/workflows/tests.yml#L205

Comment (anakin87) [reply components/file_converters/unstructured_fileconverter/docker-compose.yml]: done!

File: .github/workflows/components_unstructured_fileconverter.yml [hunk 3]
```diff
@@ -0,0 +1,59 @@
...
+      fail-fast: false
+      matrix:
+        os: [ubuntu-latest]
+        python-version: ["3.8", "3.9", "3.10", "3.11"]
+    services:
+      unstructured-api:
+        image: "quay.io/unstructured-io/unstructured-api:latest"
+        ports:
+          - 8000:8000
+        options: >-
```
[the remark this answers is not in the document]
Comment (anakin87) [reply .github/workflows/components_unstructured_fileconverter.yml:35]: Copied from [Github docs](https://docs.github.com/en/actions/using-containerized-services/creating-redis-service-containers#running-jobs-in-containers)

[the remark this answers is not in the document]
Comment (masci) [reply .github/workflows/components_unstructured_fileconverter.yml:35]: TIL it's a `strip` chomping char https://yaml.org/spec/1.2-old/spec.html#chomping//

**Review by masci — APPROVED** [review]
🚢

**Pull request merged**